### PR TITLE
feat(pie-docs): DSW-1471 add pie-button docs

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)

- filtering svg when reviewing might be helpful.

## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: .changeset/funny-flies-greet.md
Comment (ashleynolan) [inline .changeset/funny-flies-greet.md:5]:
```suggestion
[Added] - PieButton docs to pie.design
```

**Comment by ashleynolan** [comment]
@raoufswe Would you be able to break this PR down a bit into sections of the page potentially? It's getting pretty big and it just makes it quite a big task to review

**Comment by LTurns** [comment]
> @raoufswe Would you be able to break this PR down a bit into sections of the page potentially? It's getting pretty big and it just makes it quite a big task to review

On this, spoke to raouf yday and we decided to put everything in one PR so Naz could review all the content together in one review. Then I think once we've got design approval we'll break it up into smaller chunks so it's easy for engineers to review 😄

File: apps/pie-docs/src/components/button/style/destructive-colour-tokens.json
Comment (LTurns) [reply apps/pie-docs/src/components/button/style/destructive-colour-tokens.json:1]: yeah agree it's worth discussing to avoid excess overhead - I'll ask Naz about it when she's back and maybe we can have a meeting to discuss how best to implement. Thanks for bringing up 👍

Comment (ashleynolan) [reply apps/pie-docs/src/components/button/style/destructive-colour-tokens.json:1]: That'd be great. Just want to make sure we're not making a tonne of work for ourselves, especially as we'd be listing these out across every component.

**Review by FayeCarter — APPROVED** [review]

File: apps/pie-docs/src/components/button/style/destructive-colour-tokens.json
Comment (raoufswe) [reply apps/pie-docs/src/components/button/style/destructive-colour-tokens.json:1]: we decided to hide the style tab for now

**Review by LTurns — APPROVED** [review]

**Pull request merged**
